Bazel 9.0.0 compatibility

Between running CI on PRs and merging them into main, Bazel made a 9.0.0 release.
This causes our `cftests-nonjunit` tests to fail:

````
make[1]: Entering directory '/home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample'
bazelisk run example > Out.txt 2>&1 || true
grep -qF "BazelExample.java:25: error: [assignment.type.incompatible] incompatible types in assignment." Out.txt || (echo "FAILURE.  Here is file Out.txt:" && cat Out.txt && echo "End of file Out.txt." && false)
FAILURE.  Here is file Out.txt:
2026/01/20 23:21:16 Downloading https://releases.bazel.build/9.0.0/release/bazel-9.0.0-linux-x86_64...
Extracting Bazel installation...
Starting local Bazel server (9.0.0) and connecting to it...
INFO: Invocation ID: 247cf8b5-6923-4d2e-8e10-f3c6118af77f
Computing main repo mapping: 
Computing main repo mapping: 
WARNING: For repository 'rules_jvm_external', the root module requires module version rules_jvm_external@6.6, but got rules_jvm_external@6.7 in the resolved dependency graph. Please update the version in your MODULE.bazel or set --check_direct_dependencies=off
Loading: 
Loading: 0 packages loaded
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:1:12: //:example: no such attribute 'srcs' in 'java_binary' rule
make[1]: *** [Makefile:11: all] Error 1
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:1:12: //:example: no such attribute 'main_class' in 'java_binary' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:1:12: //:example: no such attribute 'deps' in 'java_binary' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:1:12: //:example: no such attribute 'plugins' in 'java_binary' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:1:12: //:example: no such attribute 'javacopts' in 'java_binary' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:13:12: //:nullness: no such attribute 'srcs' in 'java_plugin' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:13:12: //:nullness: no such attribute 'processor_class' in 'java_plugin' rule
ERROR: /home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample/BUILD:13:12: //:nullness: no such attribute 'deps' in 'java_plugin' rule
WARNING: Target pattern parsing failed.
ERROR: Skipping 'example': Error evaluating 'example': error loading package '': Package '' contains errors
ERROR: Error evaluating 'example': error loading package '': Package '' contains errors
INFO: Elapsed time: 4.350s
INFO: 0 processes.
ERROR: Build did NOT complete successfully
ERROR: Build failed. Not running target
End of file Out.txt.
make[1]: Leaving directory '/home/runner/work/checker-framework/checker-framework/docs/examples/BazelExample'
make: Leaving directory '/home/runner/work/checker-framework/checker-framework/docs/examples'

> Task :checker:exampleTests FAILED
gradle/actions: Writing build results to /home/runner/work/_temp/.gradle-actions/build-results/__run-1768951222243.json

[Incubating] Problems report is available at: file:///home/runner/work/checker-framework/checker-framework/build/reports/problems/problems-report.html
make: *** [Makefile:16: compile] Error 2

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checker:exampleTests'.
> Process 'command 'make'' finished with non-zero exit value 2

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./

BUILD FAILED in 59s
29 actionable tasks: 8 executed, 8 from cache, 13 up-to-date
Error: Process completed with exit code 1.
````

Tasks:

- [ ] Fix the Bazel version we use in CI so that random upgrades won't happen
- [ ] Fix the build to work with Bazel 9.0.0

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -26,6 +26,7 @@ jobs:
         java_version: [21]
     env:
       JAVA_VERSION: ${{ matrix.java_version }}
+      USE_BAZEL_VERSION: "8.5.1"
     steps:
     - uses: actions/checkout@v6
     - name: Set up JDK ${{ matrix.java_version }}
@@ -177,6 +178,7 @@ jobs:
 
     env:
       JAVA_VERSION: ${{ matrix.java.version }}
+      USE_BAZEL_VERSION: "8.5.1"
     continue-on-error: ${{ matrix.java.experimental }}
     steps:
     - name: Check out sources

